grakn-engine/src/main/java/ai/grakn/engine/session/GraqlSession.java

```diff
@@ -218,7 +218,7 @@ void receiveQuery(Json json) {
                 String queryString = queryStringBuilder.toString();
                 queryStringBuilder = new StringBuilder();
 
-                queries = graph.graql().infer(infer).materialise(materialise).parseList(queryString);
+                queries = graph.graql().infer(infer).materialise(materialise).parseList(queryString).collect(toList());
```

Comment:
Since you serialise the stream into a list I am not entirely sure you saving much memory (or any) in this one case.